text/0474-event-bridge-scheduler-l2.md

```diff
@@ -0,0 +1,569 @@
+
+
+## Internal FAQ
+
+> The goal of this section is to help decide if this RFC should be implemented.
+> It should include answers to questions that the team is likely ask. Contrary
+> to the rest of the RFC, answers should be written "from the present" and
+> likely discuss design approach, implementation plans, alternative considered
+> and other considerations that will help decide if this RFC should be
+> implemented.
+
+### Why are we doing this?
+
+> What is the motivation for this change?
+
+The [tracking Github issue for the module](https://github.com/aws/aws-cdk/issues/23394) has
+the 19 +1s indicating that customers want L2 CDK Construct support for this service.
+
+Here is how L2 CDK constructs for Event Scheduler can improve your AWS CDK development experience:
+
+1. CDK L2 Constructs can simplify the process of setting up permissions for invoking targets, such as AWS Lambda functions, by automatically configuring required permissions for the targets.
+2. Encrypt data by providing pre-configured encryption settings that you can use as part of your AWS infrastructure as code. This can make it easier to ensure that your data is encrypted at rest and in transit, and can help you meet compliance requirements.
+3. Add metrics and permissions to your AWS EventBridge schedules. You can use constructs to define CloudWatch alarms that monitor schedule activity, and IAM permissions that control access to schedules. This can make it easier to manage and monitor your schedules, as well as provide more granular control over who can modify them.
+4. Constructs provide pre-configured targets for common AWS services, such as Lambda functions and SQS queues, as well as a universal target that can be used to connect to other AWS services. 
+
+### Why should we _not_ do this?
+
+> Is there a way to address this use case with the current product? What are the
+> downsides of implementing this feature?
+
+
+We are not confident that the service API is fully set in stone and implementing an L2 on
+top of the current L1 may be setting us up for changes in the future. We are reaching out
+to the service team to get their input and plans for the service. 
+
+It’s a large effort (X devs * 1 week) to invest in a module when we have other pressing
+projects. However, the bulk of the effort has been spent already since we have fairly
+robust prototypes already implemented.
+
+### What is the technical solution (design) of this feature?
+
+
+```mermaid
+classDiagram
+    Expression <|-- Schedule : scheduleExpression
+    Key <|-- Schedule : key
+    ScheduleTargetBase <|-- Schedule : target
+    Group <|-- Schedule : group
+    Input <|-- ScheduleTargetBase : input
+    Targets <.. ScheduleTargetBase : implements
+
+    class Schedule {
+        + String name
+        + Duration flexibleTimeWindow 
+        + Date startDate
+        + Date endDate
+        + String scheduleTimeZone 
```

Comment:
`String` should be replaced to `TimeZone` :)

Reply:
Thanks, will fix in the next commit